algoliasearch/account_client.go

```diff
@@ -0,0 +1,123 @@
+package algoliasearch
+
+import "fmt"
+
+type accountClient struct{}
+
+func NewAccountClient() AccountClient {
+	return &accountClient{}
+
+}
+
+func (a *accountClient) CopyIndex(src, dst Index) (err error) {
+	if src.GetAppID() == dst.GetAppID() {
+		return SameAppIDErr
+	}
+
+	if _, err := dst.GetSettings(); err == nil {
+		return IndexAlreadyExistsErr
+	}
+
+	// Copy objects
```

Comment:
IMO: The objects should be the last thing to add. Why? Because if you add the objects now, when you `rules` a big re-indexation will happen at Algolia side, when you add `synonyms` a big re-indexation will happen at Algolia side and also when you add `settings` a big re-indexation will happen at Algolia side.

A good solution may be just leave the add objects to the end of the method.